servatrice/src/servatrice_database_interface.cpp

```diff
@@ -756,4 +760,41 @@ void Servatrice_DatabaseInterface::logMessage(const int senderId, const QString
     query->bindValue(":target_id", (targetType == MessageTargetChat && targetId < 1) ? QVariant() : targetId);
     query->bindValue(":target_name", targetName);
     execSqlQuery(query);
+}
+
+bool Servatrice_DatabaseInterface::changeUserPassword(const QString &user, const QString &oldPassword, const QString &newPassword)
+{
+    if(server->getAuthenticationMethod() != Servatrice::AuthenticationSql)
+        return true;
+
+    if (!checkSql())
+        return true;
+
+    if (!usernameIsValid(user))
+        return true;
+
+    QSqlQuery *passwordQuery = prepareQuery("select password_sha512 from {prefix}_users where name = :name");
```

Comment:
The prepareQuery() method will create a sql prepared statement and save it in a cache. These prepared statemen are never freed, but will be reused when needed.